src/js/hyperschedule.js

```diff
@@ -61,7 +60,8 @@ let gShowClosedCourses = true;
 let gCourseIndex = {};
 let gSelectedCoursesIndex = {};
 let gCurrentlySorting = false;
-let gCourseSearchPagesShown = 1;
+let gCourseSearchPagesShown = courseSearchPagesShownStart;
+let gEndOfResultDisplayed = false;
```

Comment:
Why is this global? It looks like it's only used in one function.

Reply:
When update gets unnecessarily called when we reach the bottom, it can add an extra “end of results” text at the bottom.

This is a hot-fix for that. It works by remembering (globally) whether the text has been added. Note that fixing the pagination bug will allow us to remove the “is end of result already displayed?” logic altogether.

I will think about this more and maybe add some comment + open an issue to remove this after the pagination bug is fixed.

Reply:
Actually never mind. I just thought of another way to fix this.